application-analytics-ui/src/main/resources/Analytics/Code/Macros/MostViewedPages.xml

```diff
@@ -0,0 +1,822 @@
+    });
+    return row[0];
+  };
+  /**
+  * This function handles the case where you get the data unified or receive data for a specific year. The function
+  * iterates through all the data and creates new rows.
+  */
+  function createMergedYearsRows(fragment, headers, data) {
+    data.forEach((item, index) =&gt; {
+      let row = createRowFromData(headers, item, index);
+      fragment.appendChild(row);
+    });
+  };
+
+  /**
+  * The data is organized as a JSON of JSONs, where each internal JSON represents data for a specific year.
+  */
+  function createMultipleYearsRows(fragment, headers, data) {
+    for (let year in data) {
+      data[year].forEach((item, index) =&gt; {
+        // Add 'year' to each item.
+        // At some point, we might want to unify the data from different years. For the time being, I don't know how
+        // Matomo handles data from multiple years.
+        // The fragmentation is caused by matomo when you use the period argument with the value of month, year or day,
+        // you can unify the data by using period=range.
+        item.year = year;
+        let row = createRowFromData(headers, item, index);
+        fragment.appendChild(row);
+      });
+    }
+  };
+
+  function createTable(container) {
+    let table = document.createElement('table');
+    let thead = document.createElement('thead');
+    let tbody = document.createElement('tbody');
+    let fragment = document.createDocumentFragment();
+
+    table.classList.add('table');
+    table.classList.add('table-hover');
+    table.classList.add('table-striped');
+
+    let url = constructUrl(container);
+    $.getJSON(url).done(function (data) {
+      // If the response is a json of jsons then we select the one with the year.
+      let headers = Array.isArray(data)
+        ? ['title', 'visits', 'hits', 'timeSpent', 'entryNbVisits', 'entryNbActions', 'bounceRate', 'exitRate']
+        : ['year', 'title', 'visits', 'hits', 'timeSpent', 'entryNbVisits', 'entryNbActions', 'bounceRate', 'exitRate'];
+      let jsonKeys = Array.isArray(data)
+        //The keys that matomo will return
+        ? ['label', 'nb_visits', 'nb_hits', 'sum_time_spent', 'entry_nb_visits', 'entry_nb_actions', 'bounce_rate', 'exit_rate']
+        : ['year', 'label', 'nb_visits', 'nb_hits', 'sum_time_spent', 'entry_nb_visits', 'entry_nb_actions', 'bounce_rate', 'exit_rate'];
+      thead.appendChild(createHeader(headers));
+      if (Array.isArray(data)) {
+        createMergedYearsRows(fragment, jsonKeys, data);
+      } else {
+        createMultipleYearsRows(fragment, jsonKeys, data);
+      }
+      tbody.appendChild(fragment);
+      table.appendChild(thead);
+      table.appendChild(tbody);
+      container.appendChild(table);
+      }).fail(function (jqxhr, textStatus, error) {
+        let err = textStatus + ', ' + error;
+        console.log('Request Failed: ' + err);
+        new XWiki.widgets.Notification(l10n.get('error', error), 'error');
+      });
```

Comment:
These should be moved 2 spaces back

Reply:
I was referring to all 4 rows
![image](https://github.com/xwikisas/application-analytics/assets/22794181/45f466d4-bf15-4330-bc3c-c62f28e7b017)